sw_utils/consensus.py

```diff
@@ -206,30 +204,24 @@ def get_consensus_client(
 
 
 async def get_chain_finalized_head(
-    consensus_client: ExtendedAsyncBeacon, slots_per_epoch: int, finality: Finality = 'finalized'
+    consensus_client: ExtendedAsyncBeacon,
+    slots_per_epoch: int,
+    state: State = 'finalized',
 ) -> ChainHead:
     """Fetches the fork safe chain head."""
-    checkpoints = await consensus_client.get_finality_checkpoint()
-    epoch: int = int(checkpoints['data'][finality]['epoch'])
-    last_slot_id: int = (epoch * slots_per_epoch) + slots_per_epoch - 1
-    for i in range(slots_per_epoch):
-        try:
-            slot = await consensus_client.get_block(str(last_slot_id - i))
-        except ClientResponseError as e:
-            if hasattr(e, 'status') and e.status == 404:
-                # slot was not proposed, try the previous one
-                continue
-            raise e
-
-        execution_payload = slot['data']['message']['body']['execution_payload']
-        return ChainHead(
-            epoch=epoch,
-            consensus_block=last_slot_id - i,
-            execution_block=BlockNumber(int(execution_payload['block_number'])),
-            execution_ts=Timestamp(int(execution_payload['timestamp'])),
-        )
-
-    raise RuntimeError(f'Failed to fetch slot for epoch {epoch}')
+    block_data = await consensus_client.get_block(state)
+    slot = int(block_data['data']['message']['slot'])
+
+    return ChainHead(
+        epoch=slot // slots_per_epoch,
+        slot=slot,
+        block_number=BlockNumber(
+            int(block_data['data']['message']['body']['execution_payload']['block_number'])
+        ),
+        execution_ts=Timestamp(
+            int(block_data['data']['message']['body']['execution_payload']['timestamp'])
+        ),
+    )
 
 
 async def get_chain_epoch_head(
```

Comment:
`get_chain_epoch_head` should not be changed, it does not relate to block finality